### PR TITLE
Filter incoming txns with too-low fees

### PR DESCRIPTION
**Pull request by psteckler**
Filter incoming transactions with fees < 2, by examining `Transaction_pool_diff` gossips.

This change partially addresses #3830; it does not filter transactions with zero amounts, since those are useful for cancellations, IIUC.

### CODE REVIEW

**Comment by psteckler** [comment]
> Can you enforce the fee requirement when transactions are created in the cli

@enolan I added the min fee enforcement in CLI. I did not add it for the GraphQL version, because I'm not sure how to do that correctly. We can add that in a separate PR.

File: src/app/cli/src/tests/coda_worker_testnet.ml [hunk 1]
```diff
@@ -467,7 +467,7 @@ module Delegation : sig
 end = struct
   let delegate_stake ?acceptable_delay:(delay = 7) (testnet : Api.t) ~node
       ~delegator ~delegatee =
-    let fee = Currency.Fee.of_int 1 in
+    let fee = Currency.Fee.of_int 2 in
```
Comment (wu-s-john) [inline src/app/cli/src/tests/coda_worker_testnet.ml]: Shouldn't this reference the User_command.minimum_fee?

File: src/app/cli/src/tests/coda_receipt_chain_test.ml [hunk 2]
```diff
@@ -32,7 +32,7 @@ let main () =
   let receiver_pk = Public_key.compress another_account_keypair.public_key in
   let sender_sk = largest_account_keypair.private_key in
   let send_amount = Currency.Amount.of_int 10 in
-  let fee = Currency.Fee.of_int 0 in
+  let fee = Currency.Fee.of_int 2 in
```
Comment (wu-s-john) [inline src/app/cli/src/tests/coda_receipt_chain_test.ml]: Shouldn't this reference User_command.minimum_fee?

File: src/app/cli/src/init/client.ml [hunk 3]
```diff
@@ -561,21 +562,28 @@ let user_command (body_args : User_command_payload.Body.t Command.Param.t)
...
-               label
-               (User_command.to_base58_check command)
-               (Receipt.Chain_hash.to_string receipt_chain_hash) )
-           ~error:(fun e -> sprintf "%s: %s" error (Error.to_string_hum e))
-           ~join_error:Or_error.join ))
+         if Currency.Fee.( < ) fee User_command.minimum_fee then (
+           printf "Fee %d is less than the minimum, %d\n%!"
+             (Currency.Fee.to_int fee)
+             (Currency.Fee.to_int User_command.minimum_fee) ;
+           Deferred.unit )
```
Comment (wu-s-john) [inline src/app/cli/src/init/client.ml]: Maybe this should exit with a non-zero integer? The transaction did fail to send after all.

**Pull request merged**